pkg/ccl/backupccl/backup_metadata.go

```diff
@@ -246,10 +246,7 @@ func writeFilesToMetadata(
 	defer fileSST.Close()
 
 	// Sort and write all of the files into a single file info SST.
-	sort.Slice(m.Files, func(i, j int) bool {
-		cmp := m.Files[i].Span.Key.Compare(m.Files[j].Span.Key)
-		return cmp < 0 || (cmp == 0 && strings.Compare(m.Files[i].Path, m.Files[j].Path) < 0)
```

Comment:
hmm here we're doing:

`(cmp == 0 && strings.Compare(m.Files[i].Path, m.Files[j].Path) < 0)`

but in our interface we're using the spans' endkey if we can't sort based on the start key. Was this a bug?

Reply:
Oh yikes, or I am about to introduce a bug.